lib/streamlit/web/server/routes.py

```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import os
+from typing import Optional
```

Comment:
Hmmm...autoflake can solve this automatically.

Reply:
<details>

```diff
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index c949bca10..37f11fb4f 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -34,6 +34,10 @@ repos:
         args:
           - --filter-files
         exclude: ^lib/streamlit/__init__\.py$
+  - repo: https://github.com/PyCQA/autoflake
+    rev: v1.7.7
+    hooks:
+      - id: autoflake
   - repo: local
     hooks:
       # Script ./scripts/run_in_subdirectory.py was used to work around a
diff --git a/e2e/scripts/st_magic.py b/e2e/scripts/st_magic.py
index e530c6ef0..b3f847073 100644
--- a/e2e/scripts/st_magic.py
+++ b/e2e/scripts/st_magic.py
@@ -124,12 +124,10 @@ def docstrings():
     def nested():
         """Multiline docstring.
         Should not be printed."""
-        pass
 
     class Foo(object):
         """Class docstring. Should not be printed."""
 
-        pass
 
     nested()
 
diff --git a/lib/streamlit/components/v1/components.py b/lib/streamlit/components/v1/components.py
index 50f974608..f0dc48642 100644
--- a/lib/streamlit/components/v1/components.py
+++ b/lib/streamlit/components/v1/components.py
@@ -37,7 +37,6 @@ LOGGER = get_logger(__name__)
 class MarshallComponentException(StreamlitAPIException):
     """Class for exceptions generated during custom component marshalling."""
 
-    pass
 
 
 class CustomComponent:
diff --git a/lib/streamlit/errors.py b/lib/streamlit/errors.py
index 7407caa66..0b685ffdd 100644
--- a/lib/streamlit/errors.py
+++ b/lib/streamlit/errors.py
@@ -38,7 +38,6 @@ class MarkdownFormattedException(Exception):
     nicely formatted on the frontend.
     """
 
-    pass
 
 
 class UncaughtAppException(Exception):
diff --git a/lib/streamlit/hello/pages/1_Plotting_Demo.py b/lib/streamlit/hello/pages/1_Plotting_Demo.py
index 0c8ef7813..d321a9ed7 100644
--- a/lib/streamlit/hello/pages/1_Plotting_Demo.py
+++ b/lib/streamlit/hello/pages/1_Plotting_Demo.py
@@ -12,8 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import inspect
-import textwrap
 import time
 
 import numpy as np
diff --git a/lib/streamlit/hello/pages/2_Mapping_Demo.py b/lib/streamlit/hello/pages/2_Mapping_Demo.py
index d8cd334e7..eb1d0e46b 100644
--- a/lib/streamlit/hello/pages/2_Mapping_Demo.py
+++ b/lib/streamlit/hello/pages/2_Mapping_Demo.py
@@ -12,8 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import inspect
-import textwrap
 from urllib.error import URLError
 
 import pandas as pd
diff --git a/lib/streamlit/hello/pages/3_DataFrame_Demo.py b/lib/streamlit/hello/pages/3_DataFrame_Demo.py
index 8aa28008e..de6ec5cf5 100644
--- a/lib/streamlit/hello/pages/3_DataFrame_Demo.py
+++ b/lib/streamlit/hello/pages/3_DataFrame_Demo.py
@@ -12,8 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import inspect
-import textwrap
 from urllib.error import URLError
 
 import altair as alt
diff --git a/lib/streamlit/runtime/app_session.py b/lib/streamlit/runtime/app_session.py
index b07884061..25dfebcec 100644
--- a/lib/streamlit/runtime/app_session.py
+++ b/lib/streamlit/runtime/app_session.py
@@ -15,7 +15,6 @@
 import asyncio
 import sys
 import uuid
-from asyncio import AbstractEventLoop
 from enum import Enum
 from typing import TYPE_CHECKING, Callable, Dict, List, Optional, Union
 
diff --git a/lib/streamlit/runtime/caching/hashing.py b/lib/streamlit/runtime/caching/hashing.py
index acb7956c1..686d87b14 100644
--- a/lib/streamlit/runtime/caching/hashing.py
+++ b/lib/streamlit/runtime/caching/hashing.py
@@ -394,4 +394,3 @@ class _CacheFuncHasher:
 class NoResult:
     """Placeholder class for return values when None is meaningful."""
 
-    pass
diff --git a/lib/streamlit/runtime/credentials.py b/lib/streamlit/runtime/credentials.py
index 0149de6b1..a91420ac5 100644
--- a/lib/streamlit/runtime/credentials.py
+++ b/lib/streamlit/runtime/credentials.py
@@ -139,7 +139,7 @@ class Credentials(object):
             raise RuntimeError(
                 'Credentials not found. Please run "streamlit activate".'
             )
-        except Exception as e:
+        except Exception:
             if auto_resolve:
                 self.reset()
                 return self.activate(show_instructions=not auto_resolve)
diff --git a/lib/streamlit/runtime/legacy_caching/hashing.py b/lib/streamlit/runtime/legacy_caching/hashing.py
index e284fa17f..79f965da6 100644
--- a/lib/streamlit/runtime/legacy_caching/hashing.py
+++ b/lib/streamlit/runtime/legacy_caching/hashing.py
@@ -785,7 +785,6 @@ def get_referenced_objects(code, context: Context) -> List[Any]:
 class NoResult:
     """Placeholder class for return values when None is meaningful."""
 
-    pass
 
 
 class UnhashableTypeError(StreamlitAPIException):
diff --git a/lib/streamlit/runtime/scriptrunner/script_runner.py b/lib/streamlit/runtime/scriptrunner/script_runner.py
index 0a975d22b..9b909b70f 100644
--- a/lib/streamlit/runtime/scriptrunner/script_runner.py
+++ b/lib/streamlit/runtime/scriptrunner/script_runner.py
@@ -639,13 +639,11 @@ class ScriptRunner:
 class ScriptControlException(BaseException):
     """Base exception for ScriptRunner."""
 
-    pass
 
 
 class StopException(ScriptControlException):
     """Silently stop the execution of the user's script."""
 
-    pass
 
 
 class RerunException(ScriptControlException):
diff --git a/lib/streamlit/runtime/state/widgets.py b/lib/streamlit/runtime/state/widgets.py
index ef70ddae4..1abc2162a 100644
--- a/lib/streamlit/runtime/state/widgets.py
+++ b/lib/streamlit/runtime/state/widgets.py
@@ -111,7 +111,6 @@ class NoValue:
     replaces `None` with a `DeltaGenerator` (for use in non-widget elements).
     """
 
-    pass
 
 
 def register_widget(
diff --git a/lib/streamlit/source_util.py b/lib/streamlit/source_util.py
index 2c55bf242..6f5b70aa0 100644
--- a/lib/streamlit/source_util.py
+++ b/lib/streamlit/source_util.py
@@ -15,7 +15,7 @@
 import re
 import threading
 from pathlib import Path
-from typing import Any, Callable, Dict, List, Optional, Tuple, cast
+from typing import Any, Callable, Dict, Optional, Tuple, cast
 
 from blinker import Signal
 
diff --git a/lib/streamlit/watcher/polling_path_watcher.py b/lib/streamlit/watcher/polling_path_watcher.py
index 8c1139348..1b11bc657 100644
--- a/lib/streamlit/watcher/polling_path_watcher.py
+++ b/lib/streamlit/watcher/polling_path_watcher.py
@@ -14,7 +14,6 @@
 
 """A class that watches a given path via polling."""
 
-import os
 import time
 from concurrent.futures import ThreadPoolExecutor
 from typing import Callable, Optional
diff --git a/lib/streamlit/web/cli.py b/lib/streamlit/web/cli.py
index 8e33e794c..741162603 100644
--- a/lib/streamlit/web/cli.py
+++ b/lib/streamlit/web/cli.py
@@ -243,7 +243,6 @@ def _main_run(file, args=None, flag_options=None):
 @main.group("cache")
 def cache():
     """Manage the Streamlit cache."""
-    pass
 
 
 @cache.command("clear")
@@ -266,7 +265,6 @@ def cache_clear():
 @main.group("config")
 def config():
     """Manage Streamlit's config settings."""
-    pass
 
 
 @config.command("show")
@@ -305,7 +303,6 @@ def test():
 
     These commands are not included in the output of `streamlit help`.
     """
-    pass
 
 
 @test.command("prog_name")
diff --git a/lib/tests/streamlit/config_option_test.py b/lib/tests/streamlit/config_option_test.py
index 34b4a6ec3..3a37550f3 100644
--- a/lib/tests/streamlit/config_option_test.py
+++ b/lib/tests/streamlit/config_option_test.py
@@ -63,7 +63,6 @@ class ConfigOptionTest(unittest.TestCase):
         @c
         def someRandomFunction():
             """Random docstring."""
-            pass
 
         self.assertEqual("Random docstring.", c.description)
         self.assertEqual(someRandomFunction._get_val_func, c._get_val_func)
diff --git a/lib/tests/streamlit/delta_generator_test.py b/lib/tests/streamlit/delta_generator_test.py
index 525cca6e7..22d125a5c 100644
--- a/lib/tests/streamlit/delta_generator_test.py
+++ b/lib/tests/streamlit/delta_generator_test.py
@@ -320,7 +320,7 @@ class DeltaGeneratorExpanderTest(DeltaGeneratorTestCase):
     def test_nested_expanders(self):
         level1 = st.expander("level 1")
         with self.assertRaises(StreamlitAPIException):
-            level2 = level1.expander("level 2")
+            level1.expander("level 2")
 
 
 class DeltaGeneratorWithTest(DeltaGeneratorTestCase):
diff --git a/lib/tests/streamlit/elements/help_test.py b/lib/tests/streamlit/elements/help_test.py
index aeed7faf0..70c8490ed 100644
--- a/lib/tests/streamlit/elements/help_test.py
+++ b/lib/tests/streamlit/elements/help_test.py
@@ -29,7 +29,6 @@ class StHelpTest(DeltaGeneratorTestCase):
 
         def my_func(some_param, another_param=123):
             """This is the doc"""
-            pass
 
         st.help(my_func)
 
diff --git a/lib/tests/streamlit/elements/image_test.py b/lib/tests/streamlit/elements/image_test.py
index faf8a5192..fbbe148ff 100644
--- a/lib/tests/streamlit/elements/image_test.py
+++ b/lib/tests/streamlit/elements/image_test.py
@@ -320,7 +320,6 @@ class ImageProtoTest(DeltaGeneratorTestCase):
 
     def test_BytesIO_to_bytes(self):
         """Test streamlit.image.BytesIO_to_bytes."""
-        pass
 
     def test_verify_np_shape(self):
         """Test streamlit.image.verify_np_shape.
@@ -350,7 +349,6 @@ class ImageProtoTest(DeltaGeneratorTestCase):
         * float with clipping
         * int  with clipping
         """
-        pass
 
     @parameterized.expand([("P", True), ("RGBA", True), ("LA", True), ("RGB", False)])
     def test_image_may_have_alpha_channel(self, format: str, expected_alpha: bool):
diff --git a/lib/tests/streamlit/elements/legacy_data_frame_test.py b/lib/tests/streamlit/elements/legacy_data_frame_test.py
index e9a29bdf5..cf1dae667 100644
--- a/lib/tests/streamlit/elements/legacy_data_frame_test.py
+++ b/lib/tests/streamlit/elements/legacy_data_frame_test.py
@@ -41,7 +41,6 @@ class LegacyDataFrameProtoTest(unittest.TestCase):
 
     def test_marshall_data_frame(self):
         """Test streamlit.data_frame.marshall_data_frame."""
-        pass
 
     def test_is_pandas_styler(self):
         """Test streamlit.data_frame._is_pandas_styler.
@@ -49,7 +48,6 @@ class LegacyDataFrameProtoTest(unittest.TestCase):
         Need to test the following:
         * object is of type pandas.io.formats.style.Styler
         """
-        pass
 
     def test_marshall_styles(self):
         """Test streamlit.data_frame._marshall_styles.
@@ -62,7 +60,6 @@ class LegacyDataFrameProtoTest(unittest.TestCase):
           * None
           * not None
         """
-        pass
 
     def test_get_css_styles(self):
         """Test streamlit.data_frame._get_css_styles.
@@ -74,7 +71,6 @@ class LegacyDataFrameProtoTest(unittest.TestCase):
         * style has name and value
         * style does not have name and value
         """
-        pass
 
     def test_get_custom_display_values(self):
         """Test streamlit.data_frame._get_custom_display_values.
@@ -87,7 +83,6 @@ class LegacyDataFrameProtoTest(unittest.TestCase):
           * true
           * false
         """
-        pass
 
     def test_marshall_pyarrow_table_data(self):
         """Test that an error is raised when called with `pyarrow.Table` data."""
diff --git a/lib/tests/streamlit/elements/pydeck_test.py b/lib/tests/streamlit/elements/pydeck_test.py
index d44d43544..a7852c313 100644
--- a/lib/tests/streamlit/elements/pydeck_test.py
+++ b/lib/tests/streamlit/elements/pydeck_test.py
@@ -13,7 +13,6 @@
 # limitations under the License.
 
 import json
-import unittest
 from unittest import mock
 
 import pandas as pd
diff --git a/lib/tests/streamlit/file_util_test.py b/lib/tests/streamlit/file_util_test.py
index bec1266f1..dae9933f1 100644
--- a/lib/tests/streamlit/file_util_test.py
+++ b/lib/tests/streamlit/file_util_test.py
@@ -56,7 +56,7 @@ class FileUtilTest(unittest.TestCase):
         self.os_stat.return_value.st_size = 0
         with pytest.raises(util.Error) as e:
             with file_util.streamlit_read(FILENAME) as input:
-                data = input.read()
+                input.read()
         self.assertEqual(str(e.value), 'Read zero byte file: "/some/cache/file"')
 
     @patch("streamlit.file_util.get_streamlit_file_path", mock_get_path)
diff --git a/lib/tests/streamlit/layouts_test.py b/lib/tests/streamlit/layouts_test.py
index 21b9badf8..3a500b788 100644
--- a/lib/tests/streamlit/layouts_test.py
+++ b/lib/tests/streamlit/layouts_test.py
@@ -30,7 +30,7 @@ class ColumnsTest(DeltaGeneratorTestCase):
 
         all_deltas = self.get_all_deltas_from_queue()
 
-        horizontal_block = all_deltas[0]
+        all_deltas[0]
         columns_blocks = all_deltas[1:4]
         widgets = all_deltas[4:]
         # 7 elements will be created: 1 horizontal block, 3 columns, 3 markdown
@@ -51,7 +51,7 @@ class ColumnsTest(DeltaGeneratorTestCase):
 
         all_deltas = self.get_all_deltas_from_queue()
 
-        horizontal_block = all_deltas[0]
+        all_deltas[0]
         columns_blocks = all_deltas[1:4]
         widgets = all_deltas[4:]
         # 7 elements will be created: 1 horizontal block, 3 columns, 3 markdown
@@ -72,7 +72,7 @@ class ColumnsTest(DeltaGeneratorTestCase):
 
         all_deltas = self.get_all_deltas_from_queue()
 
-        horizontal_block = all_deltas[0]
+        all_deltas[0]
         columns_blocks = all_deltas[1:]
         # 4 elements will be created: 1 horizontal block, 3 columns
         self.assertEqual(len(all_deltas), 4)
@@ -84,7 +84,7 @@ class ColumnsTest(DeltaGeneratorTestCase):
     def test_columns_with_default_small_gap(self):
         """Test that it works correctly with no gap argument (gap size is default of small)"""
 
-        columns = st.columns(3)
+        st.columns(3)
 
         all_deltas = self.get_all_deltas_from_queue()
 
@@ -137,7 +137,7 @@ class ExpanderTest(DeltaGeneratorTestCase):
     def test_label_required(self):
         """Test that label is required"""
         with self.assertRaises(TypeError):
-            expander = st.expander()
+            st.expander()
 
     def test_just_label(self):
         """Test that it can be called with no params"""
@@ -155,18 +155,18 @@ class TabsTest(DeltaGeneratorTestCase):
     def test_tab_required(self):
         """Test that at least one tab is required."""
         with self.assertRaises(TypeError):
-            tabs = st.tabs()
+            st.tabs()
 
         with self.assertRaises(StreamlitAPIException):
-            tabs = st.tabs([])
+            st.tabs([])
 
     def test_only_label_strings_allowed(self):
         """Test that only strings are allowed as tab labels."""
         with self.assertRaises(StreamlitAPIException):
-            tabs = st.tabs(["tab1", True])
+            st.tabs(["tab1", True])
 
         with self.assertRaises(StreamlitAPIException):
-            tabs = st.tabs(["tab1", 10])
+            st.tabs(["tab1", 10])
 
     def test_returns_all_expected_tabs(self):
         """Test that all labels are added in correct order."""
@@ -180,7 +180,7 @@ class TabsTest(DeltaGeneratorTestCase):
 
         all_deltas = self.get_all_deltas_from_queue()
 
-        horizontal_block = all_deltas[0]
+        all_deltas[0]
         tabs_block = all_deltas[1:]
         # 6 elements will be created: 1 horizontal block, 5 tabs
         self.assertEqual(len(all_deltas), 6)
diff --git a/lib/tests/streamlit/runtime/legacy_caching/caching_test.py b/lib/tests/streamlit/runtime/legacy_caching/caching_test.py
index 6e8e89b6d..d133b0eeb 100644
--- a/lib/tests/streamlit/runtime/legacy_caching/caching_test.py
+++ b/lib/tests/streamlit/runtime/legacy_caching/caching_test.py
@@ -15,7 +15,7 @@
 """st.caching unit tests."""
 import threading
 import types
-from unittest.mock import MagicMock, Mock, patch
+from unittest.mock import Mock, patch
 
 from parameterized import parameterized
 
diff --git a/lib/tests/streamlit/runtime/runtime_test.py b/lib/tests/streamlit/runtime/runtime_test.py
index 1632a7ba4..ee1f3a3bb 100644
--- a/lib/tests/streamlit/runtime/runtime_test.py
+++ b/lib/tests/streamlit/runtime/runtime_test.py
@@ -66,19 +66,19 @@ class RuntimeSingletonTest(unittest.TestCase):
 
     def test_multiple_runtime_error(self):
         """Creating multiple Runtimes raises an error."""
-        r1 = Runtime(MagicMock())
+        Runtime(MagicMock())
         with self.assertRaises(RuntimeError):
-            r2 = Runtime(MagicMock())
+            Runtime(MagicMock())
 
     def test_instance_class_method(self):
         """Runtime.instance() returns our singleton instance."""
         with self.assertRaises(RuntimeError):
             # No Runtime: error
-            instance = Runtime.instance()
+            Runtime.instance()
 
         # Runtime instantiated: no error
         _ = Runtime(MagicMock())
-        instance = Runtime.instance()
+        Runtime.instance()
 
     def test_exists(self):
         """Runtime.exists() returns True iff the Runtime singleton exists."""
diff --git a/lib/tests/streamlit/runtime/uploaded_file_manager_test.py b/lib/tests/streamlit/runtime/uploaded_file_manager_test.py
index a228f7aef..03fae8292 100644
--- a/lib/tests/streamlit/runtime/uploaded_file_manager_test.py
+++ b/lib/tests/streamlit/runtime/uploaded_file_manager_test.py
@@ -87,7 +87,7 @@ class UploadedFileManagerTest(unittest.TestCase):
         self.mgr.remove_session_files("non-report")
 
         # Add two files with different session IDs, but the same widget ID.
-        f1 = self.mgr.add_file("session1", "widget", FILE_1)
+        self.mgr.add_file("session1", "widget", FILE_1)
         f2 = self.mgr.add_file("session2", "widget", FILE_1)
 
         self.mgr.remove_files("session1", "widget")
@@ -99,8 +99,8 @@ class UploadedFileManagerTest(unittest.TestCase):
         self.mgr.remove_session_files("non-report")
 
         # Add two files with different session IDs, but the same widget ID.
-        f1 = self.mgr.add_file("session1", "widget1", FILE_1)
-        f2 = self.mgr.add_file("session1", "widget2", FILE_1)
+        self.mgr.add_file("session1", "widget1", FILE_1)
+        self.mgr.add_file("session1", "widget2", FILE_1)
         f3 = self.mgr.add_file("session2", "widget", FILE_1)
 
         self.mgr.remove_session_files("session1")
diff --git a/lib/tests/streamlit/streamlit_test.py b/lib/tests/streamlit/streamlit_test.py
index 03105a04c..301685ae0 100644
--- a/lib/tests/streamlit/streamlit_test.py
+++ b/lib/tests/streamlit/streamlit_test.py
@@ -652,7 +652,6 @@ class StreamlitAPITest(DeltaGeneratorTestCase):
 
     def test_st_legacy_vega_lite_chart(self):
         """Test st._legacy_vega_lite_chart."""
-        pass
 
     def test_st_warning(self):
         """Test st.warning."""
diff --git a/lib/tests/streamlit/watcher/util_test.py b/lib/tests/streamlit/watcher/util_test.py
index 2c918a1ff..0e86b25c2 100644
--- a/lib/tests/streamlit/watcher/util_test.py
+++ b/lib/tests/streamlit/watcher/util_test.py
@@ -52,7 +52,7 @@ class UtilTest(unittest.TestCase):
         # could easily come back to bite us if a distracted coder tweaks the
         # implementation, I'm putting this here anyway.
         with patch("streamlit.watcher.util.open", mock_open(read_data=b"hello")) as m:
-            md5 = util.calc_md5_with_blocking_retries("foo")
+            util.calc_md5_with_blocking_retries("foo")
             m.assert_called_once_with("foo", "rb")
 
 
diff --git a/scripts/cli_regression_tests.py b/scripts/cli_regression_tests.py
index beeb7ef0b..76c0f0847 100644
--- a/scripts/cli_regression_tests.py
+++ b/scripts/cli_regression_tests.py
@@ -17,7 +17,6 @@
 import os
 import signal
 import subprocess
-import time
 from typing import Optional
 
 import pytest
diff --git a/scripts/get_latest_release_candidate.py b/scripts/get_latest_release_candidate.py
index 3b500fc14..64d020335 100644
--- a/scripts/get_latest_release_candidate.py
+++ b/scripts/get_latest_release_candidate.py
@@ -19,7 +19,6 @@
 Gets all releases from PyPI, filters for release candidates of the current streamlit version.
 If there is an existing RC, prints that RC version. If there is no RC, returns None.
 """
-import json
 import os
 
 import requests
diff --git a/scripts/pypi_nightly_create_tag.py b/scripts/pypi_nightly_create_tag.py
index 40189cd67..3e320ddd7 100755
--- a/scripts/pypi_nightly_create_tag.py
+++ b/scripts/pypi_nightly_create_tag.py
@@ -18,7 +18,6 @@
 
 Increment the version number, add a dev suffix and add todays date
 """
-import sys
 from datetime import datetime
 
 import packaging.version
diff --git a/scripts/run_all_examples.py b/scripts/run_all_examples.py
index b27022a1c..e67218e69 100755
--- a/scripts/run_all_examples.py
+++ b/scripts/run_all_examples.py
@@ -17,7 +17,6 @@
 """Runs all the scripts in the examples folder (except this one)."""
 
 import os
-import sys
 
 import click
 

```
</details>